cli/command/service/update.go

```diff
@@ -361,20 +363,47 @@ func removeItems(
 	return newSeq
 }
 
-func updateMounts(flags *pflag.FlagSet, mounts *[]mounttypes.Mount) {
+type byMountSource []mounttypes.Mount
+
+func (m byMountSource) Len() int      { return len(m) }
+func (m byMountSource) Swap(i, j int) { m[i], m[j] = m[j], m[i] }
+func (m byMountSource) Less(i, j int) bool {
+	return m[i].Source < m[j].Source
+}
+
+func updateMounts(flags *pflag.FlagSet, mounts *[]mounttypes.Mount) error {
```

Comment:
This was implemented as per rest of functions behavior in this file.
Since the [``updateService()``](https://github.com/docker/docker/blob/master/cli/command/service/update.go#L120) consumes all these function to update the container spec and using out parameter to simplifies the caller code.

Changing behavior of this function only, will look odd in this code, esp ``updaqteMounts()``  is only consumed by ``updateService()``